lib/model.js

```diff
@@ -4468,7 +4468,7 @@ function _assign(model, vals, mod, assignmentOpts) {
             rawOrder[key].push(i);
           } else if (isVirtual ||
             rawDocs[key].constructor !== val.constructor ||
-            String(rawDocs[key]._doc._id) !== String(val._doc._id)) {
+            String(rawDocs[key]._id) !== String(val._id)) {
```

Comment:
Can we expand this change to check if `rawDocs[key]` and `val` are `instanceof Document`, and, if so, use `val._doc._id`, otherwise `val._id`?

Reply:
Ok, for this would you prefer it be a ternary operator inside the else if check like this
```js
(rawDocs[key] instanceof Document ? String(rawDocs[key]._doc._id) : String(rawDocs[key]._id)) !== (val instanceof Document ? String(val._doc._id) : String(val._id))
```
which looks kind of long and hard to read or move them to constant above the if check like this
```js
const keyDocId = rawDocs[key] instanceof Document ? rawDocs[key]._doc._id : rawDocs[key]._id;
const valDocId = val instanceof Document ? val._doc._id : val._id;
```
both are fine and would do the same thing, I just don't wanna keep pushing commits since the CI would rerun

Reply:
I went for the the first approach currently, just to move things forward, if it needs refactoring please let me know